File: LeetCode/35 - Search insert position/solution.py
class Solution:
    def searchInsert(self, nums: list[int], target: int) -> int:

        # if the target is in nums then get the index of target
        if target in nums:
            return nums.index(target)

        # add target to nums, sort nums, and then get the index of target
        else:
            nums.append(target)
            nums.sort()
            return nums.index(target)

File: LeetCode/35 - Search insert position/solution.swift
class Solution {
    func searchInsert(_ nums: [Int], _ target: Int) -> Int {

        // return index value of the first occurance of target
        if nums.contains(target) {
            return nums.firstIndex(of: target)!
        } else {
            // nums is a constant, so we cannot manipulate it, so we create a mutable copy of it and manipulate that instead
            var temp_nums: [Int] = nums

            temp_nums.append(target)  // add target to nums
            temp_nums.sort()  // sort nums so that the new target value is in the correct index
            return temp_nums.firstIndex(of: target)!
        }
    }
}